src/query/test/executor/transform.go

```diff
@@ -51,34 +51,34 @@ type SinkNode struct {
 
 // Process processes and stores the last block output in the sink node
 func (s *SinkNode) Process(_ *models.QueryContext, ID parser.NodeID, block block.Block) error {
-	iter, err := block.SeriesIter()
+	iter, err := block.StepIter()
 	s.Meta = block.Meta()
 	if err != nil {
 		return err
 	}
 
-	anySeries := false
+	seriesCount := len(iter.SeriesMeta())
+	steps := iter.StepCount()
+	s.Values = make([][]float64, seriesCount)
+	for i := range s.Values {
+		s.Values[i] = make([]float64, steps)
```

Comment:
Hm should we also bulk allocate these values and take subslices?  One single bulk alloc [can be a lot quicker](https://blog.kowalczyk.info/article/u5o7/speeding-up-go-with-custom-allocators.html#799d7f00-3d2a-4ffe-9cff-309e0ec52409) than many smaller ones.

i.e. 
```go
allValues := make([]float64, seriesCount*steps)
s.Values := make([][]float64, seriesCount)
for i := 0; i < seriesCount; i++ {
  s.Values = allValues[:steps]
  allValues = allValues[steps:]
}
```

Reply:
It's a test utility so it shouldn't matter too much, but will update it